hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBScanner.java

```diff
@@ -328,6 +355,152 @@ private void processRecords(ManagedRocksIterator iterator,
+            }
+          } catch (NumberFormatException ex) {
+            err().println("LESSER or GREATER operation can be performed only on numeric values.");
+            throw new IOException("Invalid filter passed");
+          }
+        } else {
+          Map<String, Filter> subfields = fieldValue.getNextLevel();
+          if (Collection.class.isAssignableFrom(valueObject.getClass())) {
+            if (!checkFilteredObjectCollection((Collection) valueObject, subfields)) {
+              return false;
+            }
+          } else if (Map.class.isAssignableFrom(valueObject.getClass())) {
+            Map<?, ?> valueObjectMap = (Map<?, ?>) valueObject;
+            boolean flag = false;
+            for (Map.Entry<?, ?> ob : valueObjectMap.entrySet()) {
+              boolean subflag;
+              if (Collection.class.isAssignableFrom(ob.getValue().getClass())) {
+                subflag = checkFilteredObjectCollection((Collection)ob.getValue(), subfields);
+              } else {
+                subflag = checkFilteredObject(ob.getValue(), ob.getValue().getClass(), subfields);
+              }
+              if (subflag) {
+                // atleast one item in the map/list of the record has matched the filter,
+                // so record passes the filter.
+                flag = true;
+                break;
+              }
+            }
+            if (!flag) {
+              // none of the items in the map/list passed the filter => record doesn't pass the filter
+              return false;
+            }
+          } else {
+            if (!checkFilteredObject(valueObject, valueClassField.getType(), subfields)) {
+              return false;
+            }
+          }
+        }
+      } catch (NoSuchFieldException ex) {
+        err().println("ERROR: no such field: " + field);
+        exception = true;
+        return false;
+      } catch (IllegalAccessException e) {
+        err().println("ERROR: Cannot get field from object: " + field);
+        exception = true;
+        return false;
+      } catch (Exception ex) {
+        err().println("ERROR: field: " + field + ", ex: " + ex);
+        exception = true;
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean checkFilteredObjectCollection(Collection<?> valueObject, Map<String, Filter> fields)
+      throws NoSuchFieldException, IllegalAccessException, IOException {
+    for (Object ob : valueObject) {
+      if (checkFilteredObject(ob, ob.getClass(), fields)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  static Field getRequiredFieldFromAllFields(Class clazz, String fieldName) throws NoSuchFieldException {
+    List<Field> classFieldList = ValueSchema.getAllFields(clazz);
+    Field classField = null;
+    for (Field f : classFieldList) {
+      if (f.getName().equals(fieldName)) {
+        classField = f;
+        break;
+      }
+    }
+    if (classField == null) {
+      throw new NoSuchFieldException();
```

Comment:
I think we should put some message as well, which can tell which field not found in which class or so